php-checks/src/main/java/org/sonar/php/checks/CountInsteadOfEmptyCheck.java

```diff
@@ -0,0 +1,116 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.php.checks;
+
+import org.sonar.check.Rule;
+import org.sonar.php.checks.utils.type.FunctionCall;
+import org.sonar.php.checks.utils.type.TreeValues;
+import org.sonar.php.checks.utils.type.TypePredicateList;
+import org.sonar.plugins.php.api.symbols.Symbol;
+import org.sonar.plugins.php.api.tree.Tree;
+import org.sonar.plugins.php.api.tree.declaration.BuiltInTypeTree;
+import org.sonar.plugins.php.api.tree.declaration.ParameterTree;
+import org.sonar.plugins.php.api.tree.declaration.TypeTree;
+import org.sonar.plugins.php.api.tree.expression.BinaryExpressionTree;
+import org.sonar.plugins.php.api.tree.expression.ExpressionTree;
+import org.sonar.plugins.php.api.tree.expression.FunctionCallTree;
+import org.sonar.plugins.php.api.tree.expression.IdentifierTree;
+import org.sonar.plugins.php.api.tree.expression.LiteralTree;
+import org.sonar.plugins.php.api.visitors.PHPVisitorCheck;
+
+@Rule(key = "S1155")
+public class CountInsteadOfEmptyCheck extends PHPVisitorCheck {
+  private static final TypePredicateList FUNCTION_PREDICATE = new TypePredicateList(
+    new FunctionCall("count"));
+
+  private static final Tree.Kind[] COMPARE_OPERATORS = {
+    Tree.Kind.GREATER_THAN_OR_EQUAL_TO,
+    Tree.Kind.GREATER_THAN,
+    Tree.Kind.LESS_THAN_OR_EQUAL_TO,
+    Tree.Kind.LESS_THAN,
+    Tree.Kind.EQUAL_TO,
+    Tree.Kind.NOT_EQUAL_TO,
+    Tree.Kind.STRICT_EQUAL_TO,
+    Tree.Kind.STRICT_NOT_EQUAL_TO
+  };
+
+  @Override
+  public void visitFunctionCall(FunctionCallTree tree) {
+    if (isCountFunction(tree) && isInZeroCompare(tree) && isArrayVariable(tree.arguments().get(0))) {
+      context().newIssue(this, tree, "Use empty() to check whether the array is empty or not.");
+    }
+    super.visitFunctionCall(tree);
+  }
+
+  private static boolean isInZeroCompare(FunctionCallTree tree) {
+    Tree parent = tree.getParent();
+    if (!parent.is(COMPARE_OPERATORS)) {
+      return false;
+    }
+
+    BinaryExpressionTree comparisonTree = (BinaryExpressionTree) parent;
+    ExpressionTree comparedValue;
+    if (comparisonTree.leftOperand().equals(tree)) {
+      comparedValue = comparisonTree.rightOperand();
+    } else {
+      comparedValue = comparisonTree.leftOperand();
+    }
+
+    return comparedValue.is(Tree.Kind.NUMERIC_LITERAL) && ((LiteralTree) comparedValue).value().equals("0");
+  }
+
+  private boolean isCountFunction(FunctionCallTree tree) {
+    return FUNCTION_PREDICATE.test(TreeValues.of(tree, context().symbolTable())) && tree.arguments().size() == 1;
```

Comment:
What about cases where [`count`](https://www.php.net/manual/en/function.count.php) has 2 arguments? I guess that we don't want to exclude such cases.

Reply:
ah I didn't now that count() can take a second parameter. >= 1 makes more sense, yes.